x/delegation/keeper/update_native_restaking_balance.go

```diff
@@ -0,0 +1,112 @@
+package keeper
+
+import (
+	sdkmath "cosmossdk.io/math"
+	"github.com/ExocoreNetwork/exocore/x/assets/types"
+	delegationtypes "github.com/ExocoreNetwork/exocore/x/delegation/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) UpdateNativeRestakingBalance(
+	ctx sdk.Context, stakerID, assetID string, amount sdkmath.Int,
+) error {
+	// todo: check if the assetID is native retaking token
+
+	if amount.IsPositive() {
+		// If the balance increases due to the client chain PoS staking reward, the increased
+		// amount can be considered a virtual deposit event. However, the increased amount needs
+		// to be manually delegated by the staker if they want it to contribute to voting power.
+		// Of course, we can also treat it as both a virtual deposit and delegation event if we
+		// think this approach is better. In that case, we would proportionally delegate the
+		// increased amount to all operators to whom the related staker has already delegated
+		// this native token.
+		err := k.assetsKeeper.UpdateStakerAssetState(ctx, stakerID, assetID, types.DeltaStakerSingleAsset{
+			TotalDepositAmount: amount,
+			WithdrawableAmount: amount,
+		})
+		if err != nil {
+			return err
+		}
+	} else if amount.IsNegative() {
+		// If the balance decreases due to the client chain PoS slashing, the decreased amount
+		// will be slashed from the withdrawable amount first, the pending undelegation second,
+		// and the delegated share last if there is still a remaining amount that needs to be slashed.
+		// When slash from the delegated share, we can slash sequentially from all delegated operators
+		// according to the order of the operator addresses in the KV store. We can also proportionally
+		// slash from all operators to whom the related staker has already delegated, if we think
+		// this approach is better
```

Comment:
I think this approach is better.

Say that native staked asset deposited by staker S is 64 ETH of which 32 ETH is free, 16 ETH is delegated to operator A and 16 ETH is unbonding from operator B. In this scenario, staker S is running a validator on Ethereum but may not necessarily be running one on Exocore or any of its AVSs.

The native staked ETH is slashed in half to 32 ETH. This half should be applied to each of the items:
- Free deposit -> 32 ETH becomes 16 ETH
- Delegated to A -> 16 ETH becomes 8 ETH and the vote power change is applied immediately across the AVSs in which A has opted.
- Unbonding from B -> 16 ETH becomes 8 ETH and continues unbonding according to the same schedule

Reply:
I will adjust the implementation when slashing from the delegated amount.